modules/300-prometheus/monitoring/prometheus-rules/deprecation.yaml

```
@@ -30,3 +30,79 @@
+        plk_protocol_version: "1"
+        plk_create_group_if_not_exists__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        plk_grouped_by__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        summary: Deprecated Grafana panel interval has found.
+        description: |
+          Before updating to Grafana 10, it's required to rewrite an outdated expressions that uses `$interval_rv`, `interval_sx3` or `interval_sx4` to `$__rate_interval`
+          Deprecated interval found on dashboard panel `{{ $labels.dashboard }}/{{ $labels.panel }}`: `{{ $labels.interval }}`
+
+- name: d8.grafana.alert_rule_deprecation
+  rules:
+    - alert: GrafanaDashboardAlertRulesDeprecated
+      expr: |
+        d8_grafana_dashboards_deprecated_alert_rule > 0
+      labels:
+        tier: cluster
+        severity_level: "8"
+      annotations:
+        plk_markup_format: markdown
+        plk_protocol_version: "1"
+        plk_create_group_if_not_exists__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        plk_grouped_by__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        summary: Deprecated Grafana alert has found.
+        description: |
+          Before updating to Grafana 10, it's required to migrate an outdated alerts from Grafana to the external alertmanager (or exporter-alertmanager stack)
+          Deprecated alert rule found on dashboard panel `{{ $labels.dashboard }}/{{ $labels.panel }}`: `{{ $labels.alert_rule }}`
+
+- name: d8.grafana.plugin_deprecation
+  rules:
+    - alert: GrafanaDashboardPluginNeedsRevision
+      expr: |
+        d8_grafana_dashboards_outdated_plugin > 0
+      labels:
+        tier: cluster
+        severity_level: "8"
+      annotations:
+        plk_markup_format: markdown
+        plk_protocol_version: "1"
+        plk_create_group_if_not_exists__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        plk_grouped_by__d8_grafana_deprecations: "GrafanaDeprecations,tier=cluster,kubernetes=~kubernetes"
+        summary: Outdated Grafana dashboard plugin has found.
+        description: |
+          Before updating to Grafana 10, it's required to check if currently installed plugins will work correctly with Grafana 10
+          Outdated plugin found on dashboard panel `{{ $labels.dashboard }}/{{ $labels.panel }}`: `{{ $labels.plugin }}`
+          {{ if (eq $labels.plugin "flant-statusmap-panel") }}
+          Plugin "flant-statusmap-panel" is being deprecated and won't be supported in the future
+          We recommend you to migrate to the Status History plugin: https://grafana.com/docs/grafana/latest/panels-visualizations/visualizations/status-history/
+          {{ end }}
+
+- name: d8.grafana.setting_deprecation
+  rules:
+    - alert: GrafanaSettingNeedsRevision
+      expr: |
+        d8_grafana_settings_outdated_plugin > 0
```

Consider removing labels if they are not needed.
```suggestion
        max by(plugin) (d8_grafana_settings_outdated_plugin) > 0
```

---

Fixed + added message about query evaluation to the alert description